x-pack/test_serverless/api_integration/test_suites/common/reporting/datastream.ts

```diff
@@ -51,10 +51,15 @@ export default function ({ getService }: FtrProviderContext) {
         roleAuthc,
         internalReqHeader
       );
+
+      generatedReports.add(job.id);
     });
 
     after(async () => {
-      await reportingAPI.deleteAllReports(roleAuthc, internalReqHeader);
```

Comment:
This was the line that causes "failures" in monitoring deployments that are used by automated tests.
1. Stages a report job, just for the side-effect of initializing the datastream
2. Deletes the report job when the test is finished, but a reference to the job is still encoded in the pending report job task
3. Deletes the saved object which is used in the report.
4. Invalidates the user credentials which are encoded in the pending report job task

Result: when it tries to execute the report job task, it will fail.
* If the task begins after the report job is deleted, it would fail because the task runner's job is to query Reporting storage for the report job data
* If the task begins before the report job is deleted and somehow accesses the report job data, it would fail because the saved object or user credentials were deleted